src/computational_graph/graph.jl

```diff
@@ -475,9 +475,51 @@ function replace_subgraph(g::Graph, w::Graph, m::Graph)
     return g0
 end
 
+function prune_unary(g::Graph{F, W}) where {F, W}
+    if (g.operator in [Prod, Sum] && length(g.subgraph) ≈ one(F) && g.subgraph_factors[1]≈ one(F) && g.factor == 1)
+        return g.subgraph[1]
+    else
+        return g
+    end
+end
 
 
-prune_unary(g::Graph) = ((length(g.subgraph) == 1 && g.subgraph_factors[1] == 1 && g.factor == 1) ? g.subgraph[1] : g)
+function inplace_prod(g1::Graph) 
+    if (length(g1.subgraphs)==1 && length(g1.subgraphs[1].subgraphs) ==1 && (g1.operator == Prod) && (g1.subgraphs[1].operator == Prod))
```

Comment:
We do not have to require `g1.operator == Prod` here, just `g1.subgraphs[1].operator == Prod`; see this [tree diagram](https://user-images.githubusercontent.com/11780326/208202649-fe670843-dc10-4d29-bcaf-7ec4a307f2e9.png) for `inplace_prod`. O (`g1.operator`) can be arbitrary.